internal/pkg/service/stream/api/service/sink.go

```diff
@@ -434,6 +434,12 @@ func (s *service) EnableSink(ctx context.Context, d dependencies.SinkRequestScop
 }
 
 func (s *service) UndeleteSink(ctx context.Context, scope dependencies.SinkRequestScope, payload *api.UndeleteSinkPayload) (res *api.Task, err error) {
+	// If user is not admin deny access for write
+	token := scope.StorageAPIToken()
+	if token.Admin == nil || token.Admin.Role != adminRole {
+		return nil, svcerrors.NewForbiddenError(s.adminError)
+	}
```

Comment:
After debating with Martin for a while we concluded it's not worth covering with tests as it would significantly complicate the setup for testing projects. Each developer would need to create a second account, invite it as guest into all his test projects and generate a token for that user so that we could have both admin and guest token in projects.json.